applications/mne_analyze/libs/anShared/Model/annotationmodel.cpp

```diff
@@ -850,3 +853,43 @@ void AnnotationModel::setGroupColor(int iGroupIndex,
 {
     m_eventGroupColor[iGroupIndex] = groupColor;
 }
+
+//=============================================================================================================
+
+void AnnotationModel::setGroupName(int iGroupIndex,
+                                   const QString &sGroupName)
+{
+    m_mAnnotationHub[iGroupIndex]->groupName = sGroupName;
+}
+
+//=============================================================================================================
+
+const QString& AnnotationModel::getGroupName(int iGroupIndex)
```

Comment:
When returning a reference, be careful that the object being referred to does not go out of scope. So it is not legal to return a reference to local var. But you can always return a reference on a static or member variable.

Reply:
We are currently returning the name saved to a member variable, m_mAnnotationhub, is that acceptable? I'm adding a check to see if the iGroupIndex key is valid.